src/utils/packageUtils.ts

```diff
@@ -0,0 +1,976 @@
+export function getPackageIdFromAlias(packageAlias: string, project: SfProject): string {
+  const packageAliases = project.getSfProjectJson().getContents().packageAliases || {};
+  // return alias if it exists, otherwise return what was passed in
+  return packageAliases[packageAlias] || packageAlias;
+}
+/**
+ * @param stringIn pascal or camel case string
+ * @returns space delimited and lower-cased (except for 1st char) string (e.g. in "AbcdEfghIj" => "Abcd efgh ij")
+ */
+export function convertCamelCaseStringToSentence(stringIn: string): string {
+  return camelCaseToTitleCase(stringIn);
+}
+/**
+ * Given a package id, attempt to return the associated aliases from the config
+ *
+ * @param packageId string representing a package id
+ * @param project for obtaining the project config
+ * @returns an array of alias for the given id.
+ */
+export function getPackageAliasesFromId(packageId: string, project: SfProject): string[] {
+  const packageAliases = project.getSfProjectJson().getContents().packageAliases || {};
+  // check for a matching alias
+  return Object.entries(packageAliases)
+    .filter((alias) => alias[1] === packageId)
+    .map((alias) => alias[0]);
+}
+export async function findOrCreatePackage2(seedPackage: string, connection: Connection): Promise<string> {
+  const query = `SELECT Id FROM Package2 WHERE ConvertedFromPackageId = '${seedPackage}'`;
+  const queryResult = await connection.tooling.query<PackagingSObjects.Package2>(query);
+  const records = queryResult.records;
+  if (records && records.length > 1) {
+    const ids = records.map((r) => r.Id);
+    throw new Error(messages.getMessage('errorMoreThanOnePackage2WithSeed', [ids.join(', ')]));
+  }
+
+  if (records && records.length === 1) {
+    // return the package2 object
+    return records[0].Id;
+  }
+
+  // Need to create a new Package2
+  const subQuery = `SELECT Name, Description, NamespacePrefix FROM SubscriberPackage WHERE Id = '${seedPackage}'`;
+  const subscriberResult = await connection.tooling.query<PackagingSObjects.SubscriberPackage>(subQuery);
+  const subscriberRecords = subscriberResult.records;
+  if (!subscriberRecords || subscriberRecords.length <= 0) {
+    throw new Error(messages.getMessage('errorNoSubscriberPackageRecord', [seedPackage]));
+  }
+
+  const request = {
+    Name: subscriberRecords[0].Name,
+    Description: subscriberRecords[0].Description,
+    NamespacePrefix: subscriberRecords[0].NamespacePrefix,
+    ContainerOptions: 'Managed',
+    ConvertedFromPackageId: seedPackage,
+  };
+
+  const createResult = await connection.tooling.create('Package2', request);
+  if (!createResult.success) {
+    throw new Error(createResult.errors.map((e) => e.message).join('\n'));
+  }
+  return createResult.id;
+}
+export function _getPackageVersionCreateRequestApi(connection: Connection): PackageVersionCreateRequestApi {
+  return new PackageVersionCreateRequestApi({ connection });
+}
+
+export async function pollForStatusWithInterval(
```

Comment:
refactored to use PollingClient. This function still needs additional refactoring